src/utils/storeSelector.js

```diff
@@ -0,0 +1,29 @@
+export const getLocale = (context) => {
```

Comment:
```js
const getConfig = context => context && context.store && context.store.getState() && context.store.getState().config

export const getLocale = (context) => {
  const config = getConfig(context)

  if (!config || !config.locale) throw Error(/* ... */)
  return config.locale
}

// etc
```

Reply:
Even better : 

const getConfig = field => (context) => {
  if (!context.store
    || !context.store.getState()
    || !context.store.getState().config
    || !context.store.getState().config[field]) {
      throw new Error(`/ HOC k-intl / ${field} is not readable. make sure that this one is available at \`config.${field}\` on your redux store`)
  }

  return context.store.getState().config[field]
}

export const getLang = getConfig('lang')
export const getLocale = getConfig('locale')
export const getFormats = getConfig('formats')
```

Reply:
Event better

```js
const getConfig = field => (context) => {
  let state
  if (context.store) state = context.store.getState()

  if (!state || !state.config || !state.config[field]) {
      throw new Error(`/ HOC k-intl / ${field} is not readable. make sure that this one is available at \`config.${field}\` on your redux store`)
  }

  return state.config[field]
}

export const getLang = getConfig('lang')
export const getLocale = getConfig('locale')
export const getFormats = getConfig('formats')
```